File: backend/internal/models/Cagnotte.go
package models

import "gorm.io/gorm"

// Cagnotte représente une cagnotte associée à un événement
type Cagnotte struct {
	ID      uint    `gorm:"primaryKey" json:"id"`
	EventID uint    `json:"event_id"`
	Total   float64 `json:"total"`
}

// Contribution représente une contribution à une cagnotte
type Contribution struct {
	ID            uint    `gorm:"primaryKey" json:"id"`
	CagnotteID    uint    `json:"cagnotte_id"`
	ParticipantID uint    `json:"participant_id"`
	Amount        float64 `json:"amount"`
}

// MigrateCagnotte initialise la table Cagnotte
func MigrateCagnotte(db *gorm.DB) {
	db.AutoMigrate(&Cagnotte{})
}

// MigrateContribution initialise la table Contribution
func MigrateContribution(db *gorm.DB) {
	db.AutoMigrate(&Contribution{})
}

File: client/lib/core/models/cagnotte.dart
class Cagnotte {
  final int id;
  final int eventId;
  final double total;

  Cagnotte({
    required this.id,
    required this.eventId,
    required this.total,
  });

  factory Cagnotte.fromJson(Map<String, dynamic> json) {
    return Cagnotte(
      id: json['id'],
      eventId: json['event_id'],
      total: json['total'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'event_id': eventId,
      'total': total,
    };
  }
}
